### PR TITLE
Add zip and sort functionality

### PR DESCRIPTION
**Pull request by ChristianBachla**
We want to train YOLOv5 on 2014 and 2017 coco datasets. The data can now be collected into one common folder for
	- "images/train"
	- "images/val"
	- "labels/train"
	- "labels/val"

### CODE REVIEW

File: utils/general.py [hunk 1]
```diff
@@ -53,6 +54,25 @@ def download_file_blob_to_dir(directory, file_blob):
     os.remove(file_destination)
 
 
+def move_downloaded_data_to_single_directory(directories):
```
Comment (buggaarde) [inline utils/general.py]: Semantically, having a function name ending in `single_directory` where the argument is called `directories`, is confusing. Try to give names that convey the meaning of the function directly.

If that doesn't make too much sense, instead try to document what the functionality is.

File: utils/general.py [hunk 2]
```diff
@@ -53,6 +54,25 @@ def download_file_blob_to_dir(directory, file_blob):
...
+    for directory in directories:
+        if not os.path.isdir(directory + "train"):
+            os.makedirs(directory + "train")
+        if not os.path.isdir(directory + "val"):
+            os.makedirs(directory + "val")
+
+        folders = os.listdir(directory)
+        for folder in folders:
+            if folder != "train" and folder != "val":
+                if folder.startswith("train"):
```
Comment (buggaarde) [inline utils/general.py]: Can I get you to add some comments here? It seems very strange to filter out "train" only to check if it starts with "train" on the very next line.

File: utils/general.py [hunk 3]
```diff
@@ -53,6 +54,25 @@ def download_file_blob_to_dir(directory, file_blob):
...
+                    copy_tree(directory + folder, directory + "train")
```
Comment (buggaarde) [inline utils/general.py]: Use a qualified import here. From [what I can tell](https://docs.python.org/3/library/shutil.html#shutil.copytree), shutil has a `copy_tree` already, so you don't need the extra import.

File: utils/general.py [hunk 4]
```diff
@@ -53,6 +54,25 @@ def download_file_blob_to_dir(directory, file_blob):
...
+
+    for directory in directories:
+        if not os.path.isdir(directory + "train"):
+            os.makedirs(directory + "train")
+        if not os.path.isdir(directory + "val"):
+            os.makedirs(directory + "val")
+
+        folders = os.listdir(directory)
+        for folder in folders:
+            if folder != "train" and folder != "val":
```
Comment (buggaarde) [inline utils/general.py]: Checks like this can be written as 
```python
if folder not in ["train", "val"]:
    ...
```
in python.

**Review by buggaarde — CHANGES_REQUESTED** [review]

File: utils/general.py (hunk 3 above)
Comment (ChristianBachla) [reply utils/general.py]: Fixed 008c684370d75e8623bc1c74dfbc0b1f8a1dedf2

File: utils/general.py (hunk 1 above)
Comment (ChristianBachla) [reply utils/general.py]: Fixed c24645245cae0f091301b47eb829857fa13fca68

File: utils/general.py (hunk 4 above)
Comment (ChristianBachla) [reply utils/general.py]: Fixed c24645245cae0f091301b47eb829857fa13fca68

File: utils/general.py (hunk 2 above)
Comment (ChristianBachla) [reply utils/general.py]: Fixed c24645245cae0f091301b47eb829857fa13fca68

Comment (ChristianBachla) [reply utils/general.py]: Fixed c11a525c00152290ce52eea5482ba430a485990b

**Review by buggaarde — APPROVED** [review]
Go! 🇩🇰

**Pull request merged**